src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -225,5 +232,339 @@ public static String getAttributeTableName(String clsName,
+	 * @param dba MySQLAdaptor
+	 * @param skiplistDbIds List<String> -- List of Pathway DbIds. If an Event being checked is a member of these pathways, they are skipped.
+	 * @return Set<GKInstance> -- All Events in DB that are not used for manual inference.
+	 * @throws Exception -- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findEventsNotUsedForManualInference(MySQLAdaptor dba, List<String> skiplistDbIds) throws Exception {
+		Set<GKInstance> eventsNotUsedForInference = new HashSet<>();
+		Collection<GKInstance> events = dba.fetchInstancesByClass(ReactomeJavaConstants.Event);
+		for (GKInstance event : events) {
+			if (!manuallyInferred(event) && !memberSkipListPathway(event, skiplistDbIds)) {
+				eventsNotUsedForInference.add(event);
+			}
+		}
+		return eventsNotUsedForInference;
+	}
+
+	/**
+	 * Finds all Events not used for Inference, and then finds subset that are Human ReactionlikeEvents
+	 * @param dba MySQLAdaptor
+	 * @param skiplistDbIds List<String> -- List of Pathway DbIds. If an Event being checked is a member of these pathways, they are skipped.
+	 * @return Set<GKInstance> -- All Human ReactionlikeEvents that are not used for manual inference.
+	 * @throws Exception-- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findHumanReactionsNotUsedForManualInference(MySQLAdaptor dba, List<String> skiplistDbIds) throws Exception {
+		Set<GKInstance> reactionsNotUsedForManualInference = new HashSet<>();
+		for (GKInstance event : findEventsNotUsedForManualInference(dba, skiplistDbIds)) {
+			// Filter for Human ReactionlikeEvents
+			if (event.getSchemClass().isa(ReactomeJavaConstants.ReactionlikeEvent)
+					&& isHumanDatabaseObject(event)) {
+
+				reactionsNotUsedForManualInference.add(event);
+			}
+		}
+		return reactionsNotUsedForManualInference;
+	}
+
+	/**
+	 * Finds all PhysicalEntities that exist in a ReactionlikeEvent's inputs, outputs, catalysts and regulations.
+	 * @param reaction GKInstance -- ReactionlikeEvent that will be searched for PhysicalEntities.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntitys in ReactionlikeEvent
+	 * @throws Exception -- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findAllPhysicalEntitiesInReaction(GKInstance reaction) throws Exception {
+		Set<GKInstance> reactionPEs = new HashSet<>();
+		reactionPEs.addAll(findAllInputAndOutputPEs(reaction));
+		reactionPEs.addAll(findAllCatalystPEs(reaction));
+		reactionPEs.addAll(findAllRegulationPEs(reaction));
+		return reactionPEs;
+	}
+
+	/**
+	 * Finds all PhysicalEntities that exist in a ReactionlikeEvent's inputs and outputs.
+	 * @param reaction GKInstance -- ReactionlikeEvent that is being searched for PhysicalEntities.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntitys in ReactionlikeEvent's inputs and outputs.
+	 * @throws Exception -- Thrown by MySQLAdaptor
+	 */
+	public static Set<GKInstance> findAllInputAndOutputPEs(GKInstance reaction) throws Exception {
```

Comment:
Should the method name changed since you have a constraint in your input/output selection? Also if this static method is used by a specific class, could it be merged into that class instead here as a generic method to be used everywhere?

Reply:
Each of these 'findPhysicalEntities' methods are used by each of the new QA checks, so they do need to be accessed generically. If I am going to switch to using InstanceUtilities methods though, the amount of code in QACheckerHelper will be reduced quite a bit